sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -80,6 +80,12 @@ abstract class Expression extends TreeNode[Expression] {
 
   def nullable: Boolean
 
+  /**
+   * Indicates whether this expression is null intolerant. If this is true,
+   * then any null input will result in null output).
+   */
+  def nullIntolerant: Boolean = true
```

Comment:
When adding new null tolerant expressions, we might hit a bug if we forget changing the default value.